editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -191,25 +167,138 @@ export function runGridRearrangeMove(
+  gridCellGlobalFrames: GridCellGlobalFrames,
+  gridTemplate: GridContainerProperties,
+  newPathAfterReparent?: ElementPath,
+): CanvasCommand[] {
+  const moveConfig = getGridMoveConfig(
+    targetElement,
+    selectedElement,
+    jsxMetadata,
+    interactionData,
+    gridCellGlobalFrames,
+    gridTemplate,
+    newPathAfterReparent,
+  )
+  if (moveConfig == null) {
+    return []
+  }
+  const {
+    newCoords,
+    gridPath,
+    targetCellData,
+    pathForCommands,
+    targetCellCoords,
+    gridTemplateColumns,
+  } = moveConfig
+  const { row, column } = newCoords
+
+  const flowElementsCount = MetadataUtils.getChildrenUnordered(jsxMetadata, gridPath).filter(
+    (child) =>
+      getGridElementPinState(child.specialSizeMeasurements.elementGridPropertiesFromProps) !==
+      'pinned',
+  ).length
+
+  // The "pure" index in the grid children for the cell under mouse
+  const possiblyReorderIndex = getGridPositionIndex({
+    row: targetCellCoords.row,
+    column: targetCellCoords.column,
+    gridTemplateColumns: gridTemplateColumns,
+  })
+
+  const updateGridControlsCommand = showGridControls(
+    'mid-interaction',
+    gridPath,
+    targetCellData?.gridCellCoordinates ?? null,
+    gridCellCoordinates(row, column),
+    possiblyReorderIndex < flowElementsCount ? 'valid' : 'invalid',
+  )
+
+  return [
+    reorderElement('always', pathForCommands, absolute(possiblyReorderIndex)),
+    deleteProperties('always', pathForCommands, [
+      PP.create('style', 'gridColumn'),
+      PP.create('style', 'gridRow'),
+      PP.create('style', 'gridColumnStart'),
+      PP.create('style', 'gridColumnEnd'),
+      PP.create('style', 'gridRowStart'),
+      PP.create('style', 'gridRowEnd'),
+    ]),
+    updateGridControlsCommand,
+  ]
+}
+
+function getGridMoveConfig(
+  targetElement: ElementPath,
+  selectedElement: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+  interactionData: DragInteractionData,
+  gridCellGlobalFrames: GridCellGlobalFrames,
+  gridTemplate: GridContainerProperties,
+  newPathAfterReparent?: ElementPath,
+) {
+  if (interactionData.drag == null) {
+    return null
+  }
+
+  const isReparent = newPathAfterReparent != null
+  const pathForCommands = newPathAfterReparent ?? targetElement // when reparenting, we want to use the new path for commands
+
+  const originalElement = MetadataUtils.findElementByElementPath(jsxMetadata, selectedElement)
+  if (originalElement == null) {
+    return null
+  }
+
+  const gridConfig = isReparent
+    ? {
+        originalCellBounds: { width: 1, height: 1 }, // when reparenting, we just put it in a single cell
+        mouseCellPosInOriginalElement: { row: 0, column: 0 },
+      }
+    : getOriginalElementGridConfiguration(gridCellGlobalFrames, interactionData, originalElement)
+  if (gridConfig == null) {
+    return null
+  }
+  const { mouseCellPosInOriginalElement } = gridConfig
+
+  const gridPath = EP.parentPath(pathForCommands)
```

Comment:
not related to this PR, but please leave a comment here that this needs to be deleted – we do not know if EP.parentPath points to a grid or not, nor should we know that

Reply:
done, I marked all usages with comments